src/App.jsx

```diff
@@ -14,45 +16,90 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABET = 'Alphabet';
+const SORT_FIELD_LENGTH = 'Length';
+const FIELD_REVERSE = 'Reverse';
+
+export const App = () => {
+  const [sortField, setSortField] = useState(null);
+  const [reverseField, setReverseField] = useState(null);
+  let visibleGoods = [...goodsFromServer];
+
+  switch (sortField) {
+    case SORT_FIELD_ALPHABET:
+      visibleGoods = visibleGoods.sort((
+        good1, good2,
+      ) => good1.localeCompare(good2));
+      break;
+    case SORT_FIELD_LENGTH:
+      visibleGoods = visibleGoods.sort((
+        good1, good2,
+      ) => good1.length - good2.length);
+      break;
+    default:
+      visibleGoods = [...goodsFromServer];
+      break;
+  }
+
+  if (reverseField) {
+    visibleGoods = visibleGoods.reverse();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info',
+            { 'is-light': sortField !== SORT_FIELD_ALPHABET })}
+          onClick={() => setSortField(sortField === SORT_FIELD_ALPHABET
+            ? null
+            : SORT_FIELD_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={cn('button is-success',
+            { 'is-light': sortField !== SORT_FIELD_LENGTH })}
+          onClick={() => setSortField(sortField === SORT_FIELD_LENGTH
+            ? null
+            : SORT_FIELD_LENGTH)}
```

Comment:
let's create separated function for this, don't repeat yourself

`const handleSort = (sortBy) => {
  if (sortField === sortBy) {
    ....
  }
...
}`